StreamChat.xcodeproj/project.pbxproj

```diff
@@ -4136,7 +4136,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				DEVELOPMENT_TEAM = EHV7XZLAHA;
 				INFOPLIST_FILE = Tests/Info.plist;
-				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.1;
```

Comment:
> This simplifies a lot the code in SnapshotVariant since there’s no need for optionals and Macros.

What kind of optionals and macros? Can you be more specific please?

Reply:
![image](https://user-images.githubusercontent.com/12814114/106490916-3a4acd80-64ae-11eb-8962-cdae27986c4c.png)
The light and dark trait collections are only supported by iOS 12+. If we have the deployment target to iOS 11.0, we need the `trait: UITraitCollection` property to be optional, because we would need a `#if` macro condition for iOS <12 and return `nil` in this case.

Reply:
Well isn't it just a problem with the fact that we chose `enum` to represent this? It's easy to skip the `userInterfaceStyle` trait collection when running on iOS11. Maybe it should rather be something like:
```swift
extension UITraitCollection { 
  static let case1: UITraitCollection = ... // small + dark (if applicable)
  ...
} 
```

Reply:
I can convert this to an option set, but the advantage I see in this case to be an enum, is that we make we always snapshot for every variant with `allCases`. If in the future we introduce a new Variant, we will always test all variants automatically. Where if we use an OptionSet, we will need to add the new option to an `all` property manually, and maybe someone can forget to add it. I bumped the version because I thought it wouldn't be that important since I don't think we will run the snapshots on iOS 11 anyways. But let me know if you think it is better for this to be an OptionSet, and put the iOS 11 support.

Reply:
We support iOS and we will eventually run snapshot tests for iOS11, too. The `allCases` benefits are minimal so I'd suggest moving from `enum` into something more flexible.